CRM/SMS/BAO/Provider.php

```diff
@@ -131,6 +137,7 @@ public static function del($providerID) {
 
     $dao = new CRM_SMS_DAO_Provider();
     $dao->id = $providerID;
+    $dao->whereAdd = "(domain_id = " .  CRM_Core_Config::domainID() . "OR domain_id IS NULL)";
```

Comment:
this is the delete function - we can remove this line again if it's problematic since it's precautionary. Not sure we could move it to the extension